2nd_HW.js

```diff
@@ -0,0 +1,168 @@
+
+//2
+3 + 2 - '1'
+//4
+/* Сначала происходит операция сложения 3+2, что даст нам 5, далее из-за того, что оператор вычитания определяется только для чисел.  Знак минус перед -'1' является унарным минус оператором , который будет фактически принуждать стринг превратиться  в число. Следовательно, '1' преобразуется в число 1, которое вычитается из 5. В итоге получим 4. */
+
+//3
+'3' + 2 - 1
+//31
+/* Как и в первом случае, сложение в JavaScript имеет ассоциативность слева направо. Это означает, что когда у нас есть пример с несколькими операторами, самый левый оператор оценивается первым. У нас это стринг. Происходит конкатенация '3' с разностью цифр 2 и 1, что даст нам 31. Не понимаю почему в итоге это не стринг.*/
+
+//4
+true + 2
+//3
+/* В JavaScript, когда оператор «плюс» помещается между числом и логическим значением, логическое значение приводится (конвертируется) в число. False -> 0, true -> 1. Другими словами вычисление происходит по принципу 1 + 2 = 3. */
+
+//5
++'10' + 1
+//11
+/* + перед стрингом, содержащим число преобразует его в число, затем происходит операция сложения 10 и 1, что даст 11 */
+
+//6
+undefined + 2
+//NaN
+/*В JS undefined + число и в обратном порядке тоже дает Not a Number, так как выражение идентично NaN + 2, что даст NaN. */
+
+//7
+null + 5
+//5
+/* null -> 0, 0 + 5 = 5 */
+
+//8
+
+true + undefined
+//NaN
+/* В JS true + undefined дает Not a Number, так как выражение идентично 1 + NaN, что даст NaN. */
+
+
+//Part 3 - If else
+
+//1
+var a = "hidden";
+
+if(a = "hidden"){
+  console.log(a = "visible");
+} else {
+  console.log(a = "hidden");
+};
+//"visible"
+
+//2
+var a = 10;
+
+if(a === 0){
+  console.log(a === 1);
+} else if(a < 0){
+  console.log(a = "less then zero");
+}else if(a > 0){
+  console.log(a *=10);
+};
+//100
+
+//3
+
+let car = { name: 'Lexus', age: 10, create: 2008, needRepair: false }
+if(car.age > 5) {
+  console.log("Need Repair");
+  car.needRepair = true;
+ }else{
+  car.needRepair = false;
+};
+console.log(car);
+
+//4
+ let item = { name: 'Intel core i7', price: '100$', discount: '15%' }
+ if(item.discount !== undefined ) {
+   item.priceWithDiscount = parseInt(item.price) * (100 - parseInt(item.discount))/100;
+   console.log(`Your price is ${item.priceWithDiscount}US$`);
+ } else{
+   console.log(`Your price is ${item.price}US$`);
+ };
+
+//5
+let product = {
+name: "Яблоко",
+price: "25$"
+}
+  let min = 10;
+  let max = 20;
+
+if(((parseInt(product.price)) >= min) && ((parseInt(product.price)) <= max)){
```

Comment:
you could write `let priceValue = parseInt(product.price); if(((priceValue >= min) && ((priceVAlue <= max)) ...` and avoid using parseInt function twice